### PR TITLE
Feature/lcz manager gps offset

### PR DESCRIPTION
**Pull request by MishkaMN**
# PR Details
## Description

Integrated python node that adds offset to gps pose as localization_manager mode. Original code in python here https://github.com/usdot-fhwa-stol/carma-platform/pull/1875

 

## Related Issue
https://github.com/usdot-fhwa-stol/carma-platform/issues/1940

## Motivation and Context
See above

## How Has This Been Tested?
Local integration tested

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [X] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: localization_manager/src/localization_manager_node.cpp [hunk 1]
```diff
@@ -117,6 +126,37 @@ namespace localization_manager
...
+        {"x_offset", config_.x_offset},
+        {"y_offset", config_.y_offset},
+        {"z_offset", config_.z_offset}}, parameters);
+        
+        auto error_2 = update_params<int>(
+        {{"auto_initialization_timeout", config_.auto_initialization_timeout},
+        {"gnss_only_operation_timeout", config_.gnss_only_operation_timeout},
+        {"gnss_data_timeout", config_.gnss_data_timeout},
+        {"sequential_timesteps_until_gps_operation", config_.sequential_timesteps_until_gps_operation},
+        {"localization_mode", config_.localization_mode}}, parameters);
```
Comment (msmcconnell) [inline localization_manager/src/localization_manager_node.cpp:146]: I don't think we can guarantee the state machine works if the localization mode is switch mid run. Probably best to remove from this callback.

**Review by msmcconnell — CHANGES_REQUESTED** [review]

File: localization_manager/config/parameters.yaml [hunk 2]
```diff
@@ -39,4 +39,10 @@ sequential_timesteps_until_gps_operation: 5
 # 2 - AUTO select between NDT and GNSS with GNSS timeout
 # 3 - AUTO select between NDT and GNSS without GNSS timeout
 # 4 - GNSS only with NDT initialization
-localization_mode: 4
\ No newline at end of file
+# 5 - GNSS only with fixed offset
+localization_mode: 4
+
+# Fixed offset to use when GNSS only with fixed offset is being used (Mode 5)
```
Comment (msmcconnell) [inline localization_manager/config/parameters.yaml:45]: specify units and frame (map)

**Review by msmcconnell — CHANGES_REQUESTED** [review]

File: localization_manager/src/localization_manager_node.cpp (hunk 1 above)
Comment (MishkaMN) [reply localization_manager/src/localization_manager_node.cpp:146]: removed it

File: localization_manager/config/parameters.yaml (hunk 2 above)
Comment (MishkaMN) [reply localization_manager/config/parameters.yaml:45]: done

**Pull request merged**